src/Runtime/PyOMCompileExecutionSession.cpp

```diff
@@ -23,10 +23,14 @@ SUPPRESS_WARNINGS_POP
 
 namespace onnx_mlir {
 
+// =============================================================================
+// Constructor
+
 PyOMCompileExecutionSession::PyOMCompileExecutionSession(
     std::string inputFileName, std::string sharedLibPath, std::string flags,
```

Comment:
Do we always compile a model? Does it make sense to skip compilation when `sharedLibPath` exists? Users may want to use the same PyOMCompileExecutionSession and expect the compilation happens only once.

If the .so file is not reused, we perhaps remove it at the end of the session.

Reply:
I see what you are saying, but then we have to look into what the compile options were... 
2 solutions:
* add a flag to say "reuse if here is already a so"
* no changes:  compile the first time, and then we can use the same .so with a regular ExecutionSession.

Reply:
Option 1 looks flexible, we can enable "reuse" by default, and for debugging, not "reuse" is useful.

Reply:
ok, will do